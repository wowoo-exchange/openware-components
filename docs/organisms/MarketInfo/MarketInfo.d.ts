import * as React from 'react';
interface MarketInfoValues {
    lastTradePrice: number;
    lastTradeCurrency: string;
    hourPrice: number;
    hourPriceChange: string;
    hourValue: number;
    hourValueCurrency: string;
}
interface MarketInfoProps {
    /**
     *  By default class name 'cr-sign-in-form'
     *  This property gives an additional class name
     */
    className?: string;
    /**
     * Platform market pairs for markets dropdown
     */
    marketPairs: string[];
    /**
     *  Market values for all market pairs
     */
    marketValues: MarketInfoValues[];
}
interface MarketInfoState {
    currentValues: MarketInfoValues;
}
declare class MarketInfo extends React.Component<MarketInfoProps, MarketInfoState> {
    constructor(props: MarketInfoProps);
    render(): JSX.Element;
    private onMarketChange;
}
export { MarketInfo, MarketInfoProps, MarketInfoValues, MarketInfoState, };
