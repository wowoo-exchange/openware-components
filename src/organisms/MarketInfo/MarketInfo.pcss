.cr-market-info {
  width: auto;
  background-color: inherit;

  .cr-market-dropdown {
    display: inline-block;
    width: calc(var(--gap) * 40);
  }

  &__pairs-dropdown {
    display: inline-block;
    font-size: calc(var(--gap) * 3.6);

    .cr-dropdown__input {
      background-color: var(--market-info-dropdown-bg-color);
      border: 0;
      padding: calc(var(--gap) * 4.7) var(--gap);
      width: inherit;

      &-label {
        background-color: var(--market-info-dropdown-bg-color);
        font-size: calc(var(--gap) * 3.6);
        font-weight: bolder;
        line-height: calc(var(--gap) * 2.5);
        color: var(--market-info-dropdown-color);
      }

      &-icon {
        margin-left: calc(var(--gap) * 3);
        background-color: var(--market-info-dropdown-bg-color);

        img {
          height: calc(var(--gap) * 2);
        }
      }

      &--open {
        background-color: var(--market-info-dropdown-bg-color);
        color: var(--market-info-dropdown-color);
        padding: calc(var(--gap) * 4.7) var(--gap);
        width: inherit;
        border: 0;

        &-icon {
          padding-left: calc(var(--gap) * 5);
        }
      }
    }

    .cr-dropdown__list {
      border: 0;

      &-item {
        background-color: var(--market-info-dropdown-bg-color);
        border: 0;
        cursor: pointer;
        padding: calc(var(--gap) * 4.7) var(--gap);

        &-label {
          background-color: var(--market-info-dropdown-bg-color);
          color: var(--market-info-dropdown-color);
          font-size: calc(var(--gap) * 3.6);
          font-weight: bolder;
          line-height: calc(var(--gap) * 2.5);
        }

        &:hover {
          background-color: var(--background-dropdown);
          color: var(--base);

          .cr-dropdown__list-item-label {
            background-color: var(--background-dropdown) !important;
          }
        }
      }
    }
  }

  &__values {
    display: inline-block;

    &-item {
      display: inline-block;
      padding-left: calc(var(--gap) * 2.7);
      padding-right: calc(var(--gap) * 2.7);
      vertical-align: middle;
    }
  }
}

@media (max-width: 700px) {
  .cr-market-dropdown {
    display: block;
    width: 100% !important;

    .cr-dropdown {
      width: 100%;
    }
  }
}

@media (max-width: 460px) {
  .cr-market-info__values {
    display: block;
    width: 100%;

    .cr-market-info__values-item {
      display: block;
      padding: calc(var(--gap) * 2) 0;
      width: 100%;
    }
  }
}
