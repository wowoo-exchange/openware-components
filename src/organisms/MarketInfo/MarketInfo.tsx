import classnames from 'classnames';
import * as React from 'react';

import { HeaderItem } from '../../atoms/index';
import { Dropdown } from '../../molecules/index';

interface MarketInfoValues {
    lastTradePrice: number;
    lastTradeCurrency: string;
    hourPrice: number;
    hourPriceChange: string;
    hourValue: number;
    hourValueCurrency: string;
}

interface MarketInfoProps {
    /**
     *  By default class name 'cr-sign-in-form'
     *  This property gives an additional class name
     */
    className?: string;
    /**
     * Platform market pairs for markets dropdown
     */
    marketPairs: string[];
    /**
     *  Market values for all market pairs
     */
    marketValues: MarketInfoValues[];
}

interface MarketInfoState {
    currentValues: MarketInfoValues;
}

class MarketInfo extends React.Component<MarketInfoProps,MarketInfoState> {
    constructor(props: MarketInfoProps) {
        super(props);
        this.state = {
          currentValues: props.marketValues[0],
        };
    }

    public render() {
        const { currentValues } = this.state;

        const { marketPairs, className } = this.props;

        const cx = classnames('cr-market-info',className);

        const changeMarket = (selected: number) => this.onMarketChange(selected);

        return (
            <nav className={cx}>
              <div className="cr-market-dropdown">
                <Dropdown className="cr-market-info__pairs-dropdown" list={marketPairs} onSelect={changeMarket}/>
              </div>
              <div className="cr-market-info__values">
                <HeaderItem className="cr-market-info__values-item" label="Last trade price" amount={currentValues.lastTradePrice} currency={currentValues.lastTradeCurrency}/>
                <HeaderItem className="cr-market-info__values-item" label="24 hour price" amount={currentValues.hourPrice} sign={currentValues.hourPriceChange}/>
                <HeaderItem className="cr-market-info__values-item" label="24 hour value" amount={currentValues.hourValue} currency={currentValues.hourValueCurrency}/>
              </div>
            </nav>
        );
    }

    private onMarketChange = (index: number) => {
        const { marketValues } = this.props;

        this.setState({
            currentValues: marketValues[index],
        });
    }
}

export {
    MarketInfo,
    MarketInfoProps,
    MarketInfoValues,
    MarketInfoState,
};
