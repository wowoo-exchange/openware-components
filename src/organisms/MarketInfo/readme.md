Market information header example:
```js
const testPairs = [
    'BTC/USDT',
    'USD/ETH',
    'EUR/USDT',
];

const testValues = [
    {
      lastTradePrice: 9168.2,
      lastTradeCurrency: 'usdt',
      hourPrice: 10.05,
      hourPriceChange: '+',
      hourValue: 2.345,
      hourValueCurrency: 'btc',
    },
    {
      lastTradePrice: 2468.2,
      lastTradeCurrency: 'eth',
      hourPrice: 9.05,
      hourPriceChange: '-',
      hourValue: 3.565,
      hourValueCurrency: 'usd',
    },
    {
      lastTradePrice: 5364.2,
      lastTradeCurrency: 'usdt',
      hourPrice: 19.05,
      hourPriceChange: '+',
      hourValue: 7.265,
      hourValueCurrency: 'eur',
    }, 
];


<MarketInfo marketPairs={testPairs} marketValues={testValues}/>
```