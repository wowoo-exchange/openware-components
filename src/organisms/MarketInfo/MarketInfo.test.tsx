import { shallow } from 'enzyme';
import * as React from 'react';
// import { HeaderItem } from '../../atoms/HeaderItem/HeaderItem';
import { MarketInfo, MarketInfoProps } from './MarketInfo';


const defaultProps: MarketInfoProps = {
    marketPairs: [
        'BTC/USDT',
        'EUR/USDT',
    ],

    marketValues: [
        {
          lastTradePrice: 9168.2,
          lastTradeCurrency: 'usdt',
          hourPrice: 10.05,
          hourPriceChange: '+',
          hourValue: 2.345,
          hourValueCurrency: 'btc',
        },
        {
          lastTradePrice: 5364.2,
          lastTradeCurrency: 'usdt',
          hourPrice: 19.05,
          hourPriceChange: '+',
          hourValue: 7.265,
          hourValueCurrency: 'eur',
        },
    ],
};

const setup = (props: Partial<MarketInfoProps> = {}) =>
    shallow(<MarketInfo {...{ ...defaultProps, ...props }} />);

describe('MarketInfo', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-market-info')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should change market values on select', () => {
        const wrapper = setup();
        wrapper.props().children[0].props.children.props.onSelect(1);
        expect(wrapper.props().children[1].props.children[0].props.amount).toEqual(defaultProps.marketValues[1].lastTradePrice);
    });
});
