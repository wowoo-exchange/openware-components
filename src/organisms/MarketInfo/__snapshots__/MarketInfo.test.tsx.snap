// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketInfo should render 1`] = `
<nav
  className="cr-market-info"
>
  <div
    className="cr-market-dropdown"
  >
    <Dropdown
      className="cr-market-info__pairs-dropdown"
      list={
        Array [
          "BTC/USDT",
          "EUR/USDT",
        ]
      }
      onSelect={[Function]}
    />
  </div>
  <div
    className="cr-market-info__values"
  >
    <HeaderItem
      amount={9168.2}
      className="cr-market-info__values-item"
      currency="usdt"
      label="Last trade price"
    />
    <HeaderItem
      amount={10.05}
      className="cr-market-info__values-item"
      label="24 hour price"
      sign="+"
    />
    <HeaderItem
      amount={2.345}
      className="cr-market-info__values-item"
      currency="btc"
      label="24 hour value"
    />
  </div>
</nav>
`;
